python/tank/platform/engine.py

```diff
@@ -54,6 +54,69 @@
 # std core level logger
 core_logger = LogManager.get_logger(__name__)
 
+
+class HostInfo(LooseVersion):
```

Comment:
This feels a bit overkill to me. The DCCs have very different versioning conventions. Feels like maybe just `host_name` and `host_version` properties would be sufficient.